source/postprocess/max_depth_field.cc

```diff
@@ -0,0 +1,131 @@
+
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    MaxDepthField<dim>::execute (TableHandler &statistics)
+    {
+      // create a quadrature formula based on the compositional element alone.
+      // be defensive about determining that a compositional field actually exists
+      AssertThrow(this->introspection().base_elements.compositional_fields != numbers::invalid_unsigned_int,
+                  ExcMessage("This postprocessor cannot be used without compositional fields."));
+      const QGauss<dim> quadrature_formula(this->get_fe().base_element(this->introspection().base_elements.compositional_fields).degree + 1);
+
+      const unsigned int n_q_points = quadrature_formula.size();
+
+      FEValues<dim> fe_values (this->get_mapping(),
+                               this->get_fe(),
+                               quadrature_formula,
+                               update_values   |
+                               update_quadrature_points |
+                               update_JxW_values);
+
+      std::vector<double> compositional_values(n_q_points);
+      std::vector<Point<dim> > position_values(n_q_points);
+
+      std::vector<double> local_max_depth(this->n_compositional_fields(), std::numeric_limits<double>::min());
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned())
+          {
+            fe_values.reinit (cell);
+
+            position_values = fe_values.get_quadrature_points();
+
+            for (unsigned int c = 0; c < this->n_compositional_fields(); ++c)
+              {
+                fe_values[this->introspection().extractors.compositional_fields[c]].get_function_values(this->get_solution(), compositional_values);
+
+                for (unsigned int q = 0; q < n_q_points; ++q)
+                  {
+                    const double depth = this->get_geometry_model().depth(position_values[q]);
+                    if (compositional_values[q] >= 0.5 && depth > local_max_depth[c])
+                      local_max_depth[c] = depth;
+                  }
+              }
+          }
+
+      // compute the max over all processors
+      std::vector<double> global_max_depth(this->n_compositional_fields());
+      Utilities::MPI::max (local_max_depth,
+                           this->get_mpi_communicator(),
+                           global_max_depth);
+
+      // finally produce something for the statistics file
+      for (unsigned int c = 0; c < this->n_compositional_fields(); ++c)
+        {
+          statistics.add_value("Max depth [km] for composition " + this->introspection().name_for_compositional_index(c),
+                               global_max_depth[c] / 1000.0);
+        }
+
+      // also make sure that the other columns filled by the this object
```

Comment:
```suggestion
      // also make sure that the other columns filled by this object
```
(I fixed the other places where you have this from in #4136.)